src/surface.cpp

```diff
@@ -1169,10 +1074,24 @@ void read_surfaces(pugi::xml_node node)
       } else {
         fatal_error(fmt::format("Invalid surface type, \"{}\"", surf_type));
       }
+
+      // Check for a periodic surface
+      if (check_for_node(surf_node, "boundary")) {
+        std::string surf_bc = get_node_value(surf_node, "boundary", true, true);
+        if (surf_bc == "periodic") {
+          if (check_for_node(surf_node, "periodic_surface_id")) {
+            int i_periodic = std::stoi(get_node_value(surf_node,
+                                                      "periodic_surface_id"));
+            periodic_pairs.push_back({model::surfaces.back()->id_, i_periodic});
```

Comment:
If we were to store the pairs of values such that the lesser (or greater) value comes first, you could use a `set` and thereby avoid the elimination of duplicates further down